src/ipc/CMakeLists.txt

```diff
@@ -6,11 +6,33 @@ elseif (CONFIG_IPC_MAJOR_4)
         add_subdirectory(ipc4)
 endif()
 
-add_local_sources(sof
+set(base_files
 	ipc-common.c
 	ipc-helper.c
 )
 
+is_zephyr(it_is)
+if(it_is) ###  Zephyr ###
+
+zephyr_library_sources(
+	${base_files}
+	dma-copy.c
+)
+
+if (CONFIG_SOC_SERIES_INTEL_CAVS_V25 OR CONFIG_SOC_SERIES_INTEL_ACE)
```

Comment:
This looks strange, but it matches the existing logic, so great to have this become more visible in the cmake code. The long per-product sections in one-giant-zephyr-cmakefile easily hide these things (when we have per-platform compilation of stuff that really shouldn't be per-platform).